skills_ml/algorithms/occupation_classifiers/train.py

```diff
@@ -94,23 +123,17 @@ def create_training_set(job_postings_generator: JobPostingGeneratorType,
     """
     X = []
     y = []
-    for i, job in enumerate(job_postings_generator):
-        if target_variable.filter_func(job):
-            label = target_variable.transformer(job)
+    jp1, jp2 = tee(job_postings_generator, 2)
+    pxy = zip_longest(pipe_x.build(jp1), pipe_y.build(jp2))
```

Comment:
This will just error if there's no pipe_x or pipe_y right? I think we should create a IdentityPipeline that just passes the input through, and default do it.